src/makeCalculator.js

```diff
@@ -4,7 +4,34 @@
  * @return {object}
  */
 function makeCalculator() {
-  // write code here
+  const calculator = {
+    result: 0,
+
+    operate(callback, number) {
+      this.result = callback(this.result, number);
+
+      return this;
+    },
+    reset() {
+      this.result = 0;
+
+      return this;
+    },
+    add(a, b) {
+      return a + b;
+    },
+    subtract(a, b) {
+      return a - b;
+    },
+    multiply(a, b) {
+      return a * b;
+    },
+    divide(a, b) {
+      return b === 0 ? a : a / b;
```

Comment:
The `divide` method currently returns `a` when `b` is zero. This might not be the expected behavior. Consider throwing an error or returning a specific value to indicate division by zero, as this is a common edge case that should be handled explicitly.